fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/impl/LuceneSearchProvider.java

```diff
@@ -318,6 +318,24 @@ private void updateDimensionRow(
     }
 
     /**
+     * Replace Lucene Index with a new index.
+     *
+     * @param newLuceneIndexPath  The path of the new index
+     */
+    public void replaceIndex(String newLuceneIndexPath) {
+        luceneIndexPath = newLuceneIndexPath;
+
```

Comment:
Sadly, we can't do it this way because we have no way of tracking, between restarts, that the index path has changed.  You'll need to try to close the MMapDirectory, file system delete it and move the newLucenePath to the location of the original path.